rusk/src/lib/lib.rs

```diff
@@ -699,120 +661,33 @@ fn accept(
     ))
 }
 
-/// Executes a transaction, returning the result of the call and the gas spent.
-/// The following steps are executed:
-///
-/// 0. Pre-flight checks, i.e. the transaction gas limit must be at least the
-///    same as what is minimally charged for a transaction of its type, and the
-///    transaction must fit in the remaining block gas.
+/// Executes a transaction, returning the receipt of the call and the gas spent.
+/// The transaction is executed in the following steps are executed:
 ///
-/// 1. Call the "spend" function on the transfer contract with unlimited gas. If
-///    this fails, the transaction should be considered invalid, or unspendable,
-///    and an error is returned.
+/// 1. Call the "spend_and_execute" function on the transfer contract with
+///    unlimited gas. If this fails, an error is returned. If an error is
+///    returned the transaction should be considered unspendable/invalid, but no
+///    re-execution is required (unless block gas limit was reached).
```

Comment:
this reads to me as if no re-execution is needed when a transaction is unspendable/invalid. Is that what you mean? Seems odd, why would we want to re-execute an invalid transaction?

Reply:
That is indeed what I mean, but not of the same transaction, but of the transactions preceding the invalid one. Made clearer this time.